de-lib/src/main/java/org/iplantc/de/collaborators/client/util/UserSearchField.java

```diff
@@ -33,13 +36,36 @@
 import com.sencha.gxt.widget.core.client.ListView;
 import com.sencha.gxt.widget.core.client.form.ComboBox;
 
+import java.util.Comparator;
+
 /**
  * @author sriram
  *
  */
 public class UserSearchField implements IsWidget,
                                         UserSearchResultSelected.HasUserSearchResultSelectedEventHandlers {
 
+    class SubjectSortComparator implements Comparator<Subject> {
+
+        @Override
+        public int compare(Subject o1, Subject o2) {
+            boolean firstIsGroup = Group.GROUP_IDENTIFIER.equals(o1.getSourceId());
+            boolean secondIsGroup = Group.GROUP_IDENTIFIER.equals(o2.getSourceId());
+            if (firstIsGroup && secondIsGroup) {
+                return o1.getSubjectDisplayName().compareToIgnoreCase(o2.getSubjectDisplayName());
+            }
+            if (firstIsGroup && !secondIsGroup) {
```

Comment:
What if `o2` is a group but `o1` is not?
Should we check if `firstIsGroup != secondIsGroup` then `return firstIsGroup ? -1 : 1`?